File: epi_judge_cpp/is_list_palindromic.cc
#include "list_node.h"

bool IsLinkedListAPalindrome(shared_ptr<ListNode<int>> L) {
  // Implement this placeholder.
  return true;
}

#include "test_framework/generic_test.h"

int main(int argc, char* argv[]) {
  std::vector<std::string> args{argv + 1, argv + argc};
  std::vector<std::string> param_names{"L"};
  GenericTestMain(args, "is_list_palindromic.tsv", &IsLinkedListAPalindrome,
                  DefaultComparator{}, param_names);
  return 0;
}

File: epi_judge_java/epi/IsListPalindromic.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class IsListPalindromic {
  @EpiTest(testfile = "is_list_palindromic.tsv")

  public static boolean isLinkedListAPalindrome(ListNode<Integer> L) {
    // Implement this placeholder.
    return true;
  }

  public static void main(String[] args) {
    System.exit(GenericTest
                    .runFromAnnotations(
                        args, new Object() {}.getClass().getEnclosingClass())
                    .ordinal());
  }
}
